webviz_subsurface/_datainput/eclipse_init_io/pvt_common.py

```diff
@@ -822,6 +830,46 @@ def viscosity_unit(self, latex: bool = False) -> str:
         """
         raise NotImplementedError("This method cannot be called from the base class.")
 
+    def density(
+        self, region_index: int, ratio: np.ndarray, pressure: np.ndarray
+    ) -> np.ndarray:
+        """Args:
+            region_index: Index of the requested PVT region
+            ratio: numpy.ndarray of ratio values the data is requested for
+            pressure: A numpy.ndarray of pressure values the data is requested for
+
+        Returns:
+            A list of the density values according to the given values.
+
+        """
+        self.validate_region_index(region_index)
+
+        return self._regions[region_index].density(ratio, pressure)
+
+    def density_unit(self, latex: bool = False) -> str:
+        """Args:
+            latex: True if the unit symbol shall be returned as LaTeX, False if not.
+
+        Returns:
+            A string containing the unit symbol of the density.
+
+        Raises a NotImplementedError when called on a base class object.
+
+        """
+        raise NotImplementedError("This method cannot be called from the base class.")
+
+    def ratio_unit(self, latex: bool = False) -> str:
+        """Args:
+            latex: True if the unit symbol shall be returned as LaTeX, False if not.
+
+        Returns:
+            A string containing the unit symbol of the phase ratio (e.g. Sm³/Sm³).
+
+        Raises a NotImplementedError when called on a base class object.
+
+        """
+        raise NotImplementedError("This method cannot be called from the base class.")
```

Comment:
Do we require that all subclasses of `PVxx` implement `density`, `density_unit` and/or `ratio_unit`, and that `PVxx` is not useful directly? If so, should we consider changing these base classes to ABCs? https://docs.python.org/3/library/abc.html